crates/bevy_math/src/curve/mod.rs

```diff
@@ -258,6 +261,149 @@ pub trait Curve<T> {
         })
     }
 
+    /// Create a new [`Curve`] inverting this curve on the x-axis, producing another curve with
+    /// outputs of the same type, effectively playing backwards starting at `self.domain().end()`
+    /// and transitioning over to `self.domain().start()`. The domain of the new curve is still the
+    /// same.
+    ///
+    /// # Error
+    ///
+    /// A [`ReverseError`] is returned if this curve's domain isn't bounded
+    fn reverse(self) -> Result<ReverseCurve<T, Self>, ReverseError>
+    where
+        Self: Sized,
+    {
+        self.domain()
+            .is_bounded()
+            .then(|| ReverseCurve {
+                curve: self,
+                _phantom: PhantomData,
+            })
+            .ok_or(ReverseError::SourceDomainEndInfinite)
+    }
+
+    /// Create a new [`Curve`] repeating this curve `n` times, producing another curve with outputs
+    /// of the same type. The domain of the new curve will be bigger by a factor of `n`.
```

Comment:
```suggestion
    /// of the same type. The domain of the new curve will be bigger by a factor of `n + 1`.
```
As currently implemented, it should be `n + 1`, I think, since `count = 0` recovers the original curve.